src/openrct2/util/SawyerCoding.cpp

```diff
@@ -384,6 +387,11 @@ static void encode_chunk_rotate(uint8_t* buffer, size_t length)
 
 int32_t sawyercoding_detect_file_type(const uint8_t* src, size_t length)
 {
+    if (length < 4)
+    {
+        throw std::runtime_error("Stream is (nearly) empty!");
```

Comment:
`std::runtime_error` is for unpredictable events, while the case is very clear here. `std::length_error` seems more appropriate here.

```suggestion
        throw std::length_error("Stream is (nearly) empty!");
```

Reply:
Also don't forget to include <stdexcept> in this source file.